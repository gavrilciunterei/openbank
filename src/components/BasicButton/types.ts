export type BasicButtonProps = {
  text: string;
  onClick?: () => void;
  backgroundColor: string;
  color: string;
  disabled?: boolean;
  type?: 'button' | 'submit' | 'reset' | undefined;
  icon?: object;
};

export type StyeProps = {
  backgroundColor: string;
  color: string;
};
