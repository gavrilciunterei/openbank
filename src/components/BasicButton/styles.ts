import styled from 'styled-components';
import { StyeProps } from './types';

export const Button = styled.button<StyeProps>`
  padding: 12px;
  background-color: ${(props) => props.backgroundColor};
  color: ${(props) => props.color};

  border: 0;
  :disabled {
    background-color: gray;
  }
`;

export const ButtonDiv = styled.div`
  display: flex;
  align-items: center;
`;
