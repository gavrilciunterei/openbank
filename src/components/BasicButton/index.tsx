import React from 'react';
import { Button } from './styles';
import { BasicButtonProps } from './types';

function BasicButton({
  text,
  onClick,
  backgroundColor,
  color,
  disabled,
  type,
}: BasicButtonProps) {
  return (
    <Button
      onClick={onClick}
      backgroundColor={backgroundColor}
      color={color}
      disabled={disabled}
      type={type ? type : 'button'}
    >
      {text}
    </Button>
  );
}

export default BasicButton;
