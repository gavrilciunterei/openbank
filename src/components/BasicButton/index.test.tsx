import React from 'react';
import renderer from 'react-test-renderer';
import BasicButton from './index';

it('Test BasicButton', () => {
  const tree = renderer
    .create(
      <BasicButton text="BotÃ³n de prueba" backgroundColor="red" color="blue" />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
