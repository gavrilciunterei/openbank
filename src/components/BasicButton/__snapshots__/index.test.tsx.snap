// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test BasicButton 1`] = `
<button
  className="sc-bdvvtL fUgaqB"
  color="blue"
  type="button"
>
  <div
    className="sc-gsDKAQ hznLYe"
  >
    Bot√≥n de prueba
  </div>
</button>
`;
